packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1503,6 +1523,28 @@ export class SnapController extends BaseController<
     }
   }
 
+  /**
+   * Checks if a list of permissions are dynamic and allowed to be revoked, if they are they will all be revoked.
+   *
+   * @param snapId - The snap ID.
+   * @param permissionNames - The names of the permissions.
+   * @throws If non-dynamic permissions are passed.
+   */
+  revokeDynamicSnapPermissions(
```

Comment:
Should we just name this `revokePermissions`? Is there a need (in the long term) to make a difference between dynamic and static permissions?

Reply:
I'm not sure. I named it like this to avoid confusion with the `PermissionController`. 

If all snap permissions end up being dynamic we can probably just remove this again and use the PermissionController directly. But for now it needs to be managed by the SnapController 🤔